Make WIT iss claim optional and add note about its relation to key distribution (#96)
ietf-wg-wimse/draft-ietf-wimse-s2s-protocol: draft-ietf-wimse-s2s-protocol.md

informative:
  IANA.JOSE.ALGS: IANA.jose_web-signature-encryption-algorithms
  OpenID.Discovery:
    title: "OpenID.Discovery"
    target: https://openid.net/specs/openid-connect-discovery-1_0.html

> 

Reviewer: ```suggestion
```

thus, `Workload-Identity-Token`, `workload-identity-token`, `WORKLOAD-IDENTITY-TOKEN`,
etc., are all valid and equivalent header field names. However, case is significant in the header field value.

### A note on `iss` claim and key distribution {#wit-iss-note}

> It is RECOMMENDED that the WIT carries an `iss` claim. This specification itself does not make use of a potential `iss` claim but also carries the {{trust-domain}} in the {{workload-identifier}}. Implementations MAY include the `iss` claim in the form of a `https`-URL to facilitate key distribution via {{OpenID.Discovery}} or {{!RFC8414}} but alternative key distributions may make use of the {{trust-domain}} of the {{workload-identifier}} carried in the required `sub` claim.

Reviewer 2: ```suggestion
It is RECOMMENDED that the WIT carries an `iss` claim. This specification itself does not make use of a potential `iss` claim but also carries the trust domain in the workload identifier ({{workload-identifier}}). Implementations MAY include the `iss` claim in the form of a `https` URL to facilitate key distribution via {{OpenID.Discovery}} or {{!RFC8414}} but alternative key distribution methods may make use of the trust domain included in the workload identifier which is carried in the mandatory `sub` claim.
```

Reviewer 2: @arndt-s note that {{trust-domain}} translates in "section XXX" - it's a reference.

Author: Thank you!

> It is RECOMMENDED that the WIT carries an `iss` claim. This specification itself does not make use of a potential `iss` claim but also carries the trust domain in the workload identifier ({{workload-identifier}}). Implementations MAY include the `iss` claim in the form of a `https` URL to facilitate key distribution via {{OpenID.Discovery}} or {{!RFC8414}} but alternative key distribution methods may make use of the trust domain included in the workload identifier which is carried in the mandatory `sub` claim.

Reviewer: let's keep OIDC out of this and make the RFC8414 statement a little better qualified 
```suggestion
It is RECOMMENDED that the WIT carries an `iss` claim. This specification itself does not make use of a potential `iss` claim but also carries the trust domain in the workload identifier ({{workload-identifier}}). Implementations MAY include the `iss` claim in the form of a `https` URL to facilitate key distribution via mechanisms like the `jwks_uri` from {{!RFC8414}} but alternative key distribution methods may make use of the trust domain included in the workload identifier which is carried in the mandatory `sub` claim.
```
